### PR TITLE
Add Celery support and supporting files. Fixes #21

### LINKED ISSUE #21: Add Celery
We use it on most TBX projects, so worth adding it.

Celery 3.1+ integrates with Django out of the box, so no need for django-celery.
See http://docs.celeryproject.org/en/latest/django/first-steps-with-django.html

### PR DESCRIPTION
**Pull request by zerolab**
Added Celery support (see #21)

Follows recommendations from http://celery.readthedocs.org/en/latest/django/first-steps-with-django.html and http://www.revsys.com/12days/async-workers-celery/

Provides a note about multiple sites using the same Redis backend.

### CODE REVIEW

File: project_name/home/tasks.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+try:
+    from celery.decorators import task
+    NO_CELERY = False
+except:
+    NO_CELERY = True
+
+# However, we could have installed celery for other projects. So we will also
+# check if we have defined the BROKER_URL setting. If not then definitely we
+# haven't configured it.
+if NO_CELERY or not hasattr(settings, 'BROKER_URL'):
```
Comment (kaedroho) [inline project_name/home/tasks.py]: ``settings`` is not imported from anywhere here.

File: project_name/project_name/celery.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+"""
+
+from __future__ import absolute_import
+
+import os
+from celery import Celery
+from django.conf import settings
+
+# set the default Django settings module for the 'celery' program.
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "{{ project_name }}.settings.production")
```
Comment (kaedroho) [inline project_name/project_name/celery.py]: Could we set the default settings file to dev settings like manage.py? (production servers would set the ``DJANGO_SETTINGS_MODULE`` variable to make it use production settings anyway)

File: project_name/project_name/settings/dev.py [hunk 3]
```diff
@@ -9,6 +9,11 @@
 
 EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'
 
+# Process all tasks synchronoysly.
```
Comment (kaedroho) [inline project_name/project_name/settings/dev.py]: Typo

File: project_name/home/tasks.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+"""
+Sample tasks file.
+Will fallback to a simple task decorator that returns the original function.
+"""
+
+# The following will check to see if we can import task from celery -
+# if not then we definitely haven't installed it.
+# Taken from wagtail core
+try:
+    from celery.decorators import task
```
Comment (kaedroho) [inline project_name/home/tasks.py]: As this is in requirements.txt, this test will always succeed even if celery is not actually running.

I don't think it's useful to test for celery as the developer should be aware if their project uses celery or not.

Comment (zerolab) [inline project_name/home/tasks.py]: I took this from Wagtail core. Will remove.

**Comment by zerolab** [comment]
Ready for re-review.

**Comment by kaedroho** [comment]
Looks good to me!

**Pull request merged**